### PR TITLE
BoM Features to Pandas

### PR DESCRIPTION
**Pull request by parksjr5**
This PR includes all Four API requests for BoM features. Those requests are:

1. list the features available in BoM data
2. Show basic feature information
3. Show more detailed information
4. Get data for certain features, within certain time range

There is also the ability to get the features from the data (point 4) and put them in a pandas dataframe.

### CODE REVIEW

File: readysignal/__init__.py [hunk 1]
```diff
@@ -151,30 +166,217 @@ def auto_discover(access_token, geo_grain, date_grain, filename=None, df=None, c
...
+    :param type: list of integer(s)
+    :param start_date: start date for features
+    :param type: string in Y-m-d format
+    :param end_date: end_date for features
+    :param type: string in Y-m-d format
+    :param details: show feature details
+    :param type: boolean
+    :return: request response as json
+    """
+    # bom_features = []
```
Comment (jennank) [inline readysignal/__init__.py]: Remove commented code

File: tests/test_readysignal.py [hunk 2]
```diff
@@ -23,77 +37,172 @@ def test_list_signals():
 
     :return: list of signal ids, or error
     """
-    access_token = creds['access_token']
+    access_token = creds["access_token"]
     signals = list_signals(access_token)
 
     assert isinstance(signals, dict)
-    assert 'data' in signals.keys()
+    # assert 'data' in signals.keys()
```
Comment (jennank) [inline tests/test_readysignal.py]: Remove the comment, this should get tested. We missed removing the # after commenting these pieces out earlier.

File: tests/test_readysignal.py [hunk 3]
```diff
@@ -23,77 +37,172 @@ def test_list_signals():
...
+    file_name = "test1.csv"
     signal_to_csv(access_token, signal_id, file_name)
 
     return file_name
 
 
 def test_auto_discover():
-    access_token = creds['access_token']
+    access_token = creds["access_token"]
     # response = auto_discover(access_token, 'Country', filename='../tests/country.csv')
```
Comment (jennank) [inline tests/test_readysignal.py]: Remove commented code (from before but should still be removed)

File: README.md [hunk 4]
```diff
@@ -285,3 +305,191 @@ Set `create_custom_features=0` to prevent Ready Signal from storing the input ta
 ```python
 rs.auto_discover(access_token, geo_grain, date_grain, filename=None, df=None, create_custom_features=1)
 ```
+
+<br>
+## * Feature ID Specific*
```
Comment (jennank) [inline README.md:311]: This is not formatting as expected. I believe you need a newline after the `<br>` above

**Review by jennank — CHANGES_REQUESTED** [review]
Overall things are looking good!  A few minor changes, and then we can update the version number before merging. Re-request my review whenever you're ready!

File: tests/test_readysignal.py [hunk 5]
```diff
@@ -1,6 +1,20 @@
...
+from readysignal import (
+    connect_to_readysignal_features,
+    get_features_list,
+    show_feature,
+    show_feature_detailed,
+    get_feature_data,
+    get_feature_data_pandas,
+)
 import pandas as pd
 import time
```
Comment (jennank) [inline tests/test_readysignal.py]: unused import - please delete

File: readysignal/__init__.py (hunk 1 above)
Comment (parksjr5) [reply readysignal/__init__.py]: Waiting to hear back if there is a specific list of feature ids we should check the inputs with.

File: tests/test_readysignal.py [hunk 6]
```diff
@@ -23,83 +35,181 @@ def test_list_signals():
...
+def test_show_feature():
+    """
+    tests an API call to get list of all Bank of Mexico features
+
+    :return: dictionary of feature(s)'s basic information, or error
+    """
+    access_token = creds["access_token"]
+    feature = [317]
+    feat_show = show_feature(access_token, feature)
+    feat_dict = list(feat_show.values())[0]
```
Comment (jennank) [inline tests/test_readysignal.py:144]: Unused variable

File: tests/test_readysignal.py [hunk 7]
```diff
@@ -23,83 +35,181 @@ def test_list_signals():
...
+def test_show_feature_detailed():
+    """
+    tests an API call to get list of all Bank of Mexico features
+
+    :return: dictionary of feature(s)'s detailed information, or error
+    """
+    access_token = creds["access_token"]
+    feature = [317]
+    feat_details = show_feature_detailed(access_token, feature)
+    feat_dict = list(feat_details.values())[0]
```
Comment (jennank) [inline tests/test_readysignal.py]: Unused variable. Should there be an assert like the function above?

File: tests/test_readysignal.py [hunk 8]
```diff
@@ -23,83 +35,181 @@ def test_list_signals():
...
+
+    :return: pandas DataFrame of features and data, or error
+    """
+    access_token = creds["access_token"]
+    features = [317]
+    start_date = "2021-01-01"
+    end_date = "2021-12-31"
+    df = get_feature_data_pandas(access_token, features, start_date, end_date)
+
+    # assert isinstance(df, pd.DataFrame)
```
Comment (jennank) [inline tests/test_readysignal.py]: Commented code

**Review by jennank — CHANGES_REQUESTED** [review]

**Review by jennank — APPROVED** [review]
Everything looks great to me! Will update the version before the merge.

@baylenspringer FYI we are good to merge these updates in if you are.

**Pull request merged**